DigitalLearningSolutions.Web/Controllers/Register/RegisterAtNewCentreController.cs

```diff
@@ -0,0 +1,349 @@
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement;
+
+    [SetDlsSubApplication(nameof(DlsSubApplication.Main))]
+    [SetSelectedTab(nameof(NavMenuTab.Register))]
+    public class RegisterAtNewCentreController : Controller
+    {
+        private readonly ICentresDataService centresDataService;
+        private readonly IFeatureManager featureManager;
+        private readonly PromptsService promptsService;
+        private readonly IRegistrationService registrationService;
+        private readonly ISupervisorDelegateService supervisorDelegateService;
+        private readonly IUserService userService;
+
+        public RegisterAtNewCentreController(
+            ICentresDataService centresDataService,
+            IFeatureManager featureManager,
+            PromptsService promptsService,
+            IRegistrationService registrationService,
+            ISupervisorDelegateService supervisorDelegateService,
+            IUserService userService
+        )
+        {
+            this.centresDataService = centresDataService;
+            this.featureManager = featureManager;
+            this.promptsService = promptsService;
+            this.registrationService = registrationService;
+            this.supervisorDelegateService = supervisorDelegateService;
+            this.userService = userService;
+        }
+
+        public IActionResult Index(int? centreId = null, string? inviteId = null)
+        {
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Register", new { centreId, inviteId });
+            }
+
+            if (!CheckCentreIdValid(centreId))
+            {
+                return NotFound();
+            }
+
+            // TODO HEEDLS-899 sort out supervisor delegate stuff, this is just copied from the external registration
+            var supervisorDelegateRecord = centreId.HasValue && !string.IsNullOrEmpty(inviteId) &&
+                                           Guid.TryParse(inviteId, out var inviteHash)
+                ? supervisorDelegateService.GetSupervisorDelegateRecordByInviteHash(inviteHash)
+                : null;
+
+            if (supervisorDelegateRecord?.CentreId != centreId)
+            {
+                supervisorDelegateRecord = null;
+            }
+
+            TempData.Set(
+                new InternalDelegateRegistrationData(
+                    centreId,
+                    supervisorDelegateRecord?.ID,
+                    supervisorDelegateRecord?.DelegateEmail
+                )
+            );
+
+            return RedirectToAction("PersonalInformation");
+        }
+
+        [HttpGet]
+        [Authorize(Policy = CustomPolicies.BasicUser)]
+        [ServiceFilter(typeof(RedirectEmptySessionData<InternalDelegateRegistrationData>))]
+        public IActionResult PersonalInformation()

```

Comment:
I'm not sure what value there is to adding a unit test for these Get methods since there isn't really any logic in them. They are technically tested that the load except the Confirmation one in the UI tests though.

Reply:
Fair enough. I wasn't sure how committed we are to getting as close as possible to 100% coverage